### DISCUSSION TITLE
Can not `unwatch` storage when it is already `onInvalidated`

### DISCUSSION THREAD

**Discussion by 1natsu172** [wxt-dev/wxt#903] [Q&A] [answered]
I found this issue while investigating the `Extension context invalidated.` problem in my project. But I don't know what the right thing to do about this issue should be.……I don't even know if it is necessary to cancel the `watch` when a real invalidated occurs.

In repro, I'm only doing `unwatch` for simplicity, but I think the same problem will probably occur if some access to a storage api(e.g. `getItem`).

## Repro

- my forked repo branch based by wxt/example
  - https://github.com/1natsu172/examples/tree/repro-invalidated/examples/react-content-script-ui
    - Specific Codes
      - https://github.com/1natsu172/examples/blob/repro-invalidated/examples/react-content-script-ui/entrypoints/content/executor.ts

### Steps

1. CMDs
```
pnpm i
pnpm build
```

2. Open https://google.com
3. Click anywhere on the `body` for check work
4. Reload extension package
5. Go back to the tab and click on body one more time.

https://github.com/user-attachments/assets/806f6270-a4e1-46a4-bce7-425eadfbdae5

**Comment by 1natsu172** [comment]
If doesn't become a memory leak, I think can get away with no touch the storage listener. In that case, the practice would be this?

```
if(this.ctx.isValid) { // OR `if(browser.runtime.id) {`
  this.#unwatchFoo?.();
}
```

**Reply by aklinker1** [reply, marked as answer]
It should be fine to leave it. Listeners stop working once the context is invalidated. Not sure if they're removed or not either.

We should update storage to use `browser.storage?.xxx` when removing listeners, just to be safe

**Reply by 1natsu172** [reply]
Yeah, I had a feeling. But I wasn't sure, so I wanted to talk to someone about it. Thank you 😃
